hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommits.java

```diff
@@ -129,6 +130,52 @@ public void testThriftExceptionUnknownStateIfNotInHistoryFailureOnCommit()
         .isEqualTo(3);
   }
 
+  /**
+   * Pretends we throw a concurrent commit to allow for retry when register and overwrite an iceberg
+   * table
+   */
+  @Test
+  public void testCommitFailedExceptionOnRegisterOverwrite() {
+    Table table = catalog.loadTable(TABLE_IDENTIFIER);
+    HiveTableOperations ops = (HiveTableOperations) ((HasTableOperations) table).operations();
+
+    TableMetadata metadataV1 = ops.current();
+    String metadataV1Location = metadataV1.metadataFileLocation();
+    String metadataV1AsJson = TableMetadataParser.toJson(metadataV1);
+
+    table.updateSchema().addColumn("n", Types.IntegerType.get()).commit();
+
+    ops.refresh();
+
+    TableMetadata metadataV2 = ops.current();
+
+    assertThat(ops.current().schema().columns()).hasSize(2);
+
+    HiveTableOperations spyOps = spy(ops);
+
+    HiveCatalog spyCatalog = spy(catalog);
+    when(spyCatalog.newTableOps(TABLE_IDENTIFIER)).thenReturn(spyOps);
+    // Simulate a CommitFailedException and then allow the commit to be retried
+    doThrow(new CommitFailedException("Cannot commit: stale table metadata"))
```

Comment:
I'm not sure this is correct. This would throw the exception and call the real method correct? If you wanted to trigger this just once you need to use that mock. 

This pattern is used in other places because we want to throw an exception but secretly have the method succeed at the same time.

Reply:
Ended up not need this tests.